### ISSUE TITLE
Support let-binding for linear variables

### ISSUE THREAD

**Issue by clayrat** [idris-lang/Idris-dev#4297]
# Steps to Reproduce
```
%language LinearTypes
%default total

foo : (1 x : Int) -> ((1 i : Int) -> String) -> String
foo x f = f x

bar : (1 x : Int) -> ((1 i : Int) -> String) -> String
bar x f = let res = f x in res
```
# Expected Behavior
Both `foo` and `bar` type check (possibly, with some additional syntax, e.g. `let (1 res) = ...`.

# Observed Behavior
On Idris 1.2.0:
```
  |
8 | bar x f = let res = f x in res
  |           ~~~~~~~~~~~~~~~~~~~~
Trying to use linear name x in non-linear context
```

@Melvar on IRC had more details on how this could be implemented, unfortunately I have lost access to those logs.

**Comment by Melvar** [comment]
The `PTerm` constructor `PLet` representing true `let` already has a `RigCount` field that presumably, when set to `Rig1`, would do the right thing, but the current parser function `let_` does not parse a rigcount and always passes `RigW`, i.e. unlimited use, as that field.

**Comment by ahmadsalim** [comment]
Fixed by #4298 .
